packages/syft/src/syft/types/syft_object.py

```diff
@@ -44,14 +45,37 @@
 HIGHEST_SYFT_OBJECT_VERSION = max(supported_object_versions)
 LOWEST_SYFT_OBJECT_VERSION = min(supported_object_versions)
 
+DEFAULT_HASH_EXCLUDE_ATTRS = [
+    "syft_node_location",
+    "syft_client_verify_key",
+]
 
-class SyftBaseObject(BaseModel):
+
+class SyftHashableObject:
+    __hash_exclude_attrs__ = []
+
+    def __hash__(self) -> int:
+        return int.from_bytes(self.__sha256__(), byteorder="big")
+
+    def __sha256__(self) -> bytes:
+        self.__hash_exclude_attrs__.extend(DEFAULT_HASH_EXCLUDE_ATTRS)
+        _bytes = serialize(self, to_bytes=True, for_hashing=True)
+        return sha256(_bytes).digest()
+
+    def hash(self) -> str:
+        return self.__sha256__().hex()
+
+
+class SyftBaseObject(BaseModel, SyftHashableObject):
     class Config:
         arbitrary_types_allowed = True
 
     __canonical_name__: str  # the name which doesn't change even when there are multiple classes
     __version__: int  # data is always versioned
 
+    syft_node_location: Optional[UID]
+    syft_client_verify_key: Optional[SyftVerifyKey]
+
```

Comment:
As a headsup once we make `SyftBaseObject` as hashable and have node & client info, the following classes will be have them as well:
- APIEndpoint
- LibEndpoint
- SyftAPIData
- PreHookContext
- BaseConfig
- StoreConfig
- MongoDict
- ChangeContext
- Context

If this is specific to node/request then would suggest we don't add them to SyftBaseObject, and instead create a new sub-class and use it for the concrete ones that need these attrs.

Reply:
This is not specific to request, but most of the SyftObjects, Error and Permission objects. Yes I will be creating a sub-class, to prevent all pydantic classes have these attributes.